src/cloudant/design_document.py

```diff
@@ -97,37 +85,28 @@ def update_view(self, view_name, map_func, reduce_func=None, **kwargs):
         if view is None:
             msg = "View {0} does not exist in this design doc".format(view_name)
             raise CloudantArgumentError(msg)
-        if isinstance(view, QueryIndexView):
-            msg = 'Cannot update a query index view using this method.'
-            raise CloudantException(msg)
 
         view = View(self, view_name, map_func, reduce_func, **kwargs)
         self.views.__setitem__(view_name, view)
 
     def delete_view(self, view_name):
         """
         Removes an existing MapReduce view definition from the locally cached
-        DesignDocument View dictionary.  To delete a JSON query index
-        use :func:`~cloudant.database.CloudantDatabase.delete_query_index`
-        instead.  A CloudantException is raised if an attempt to delete a
-        QueryIndexView (JSON query index) using this method is made.
+        DesignDocument View dictionary.
 
         :param str view_name: Name used to identify the View.
         """
         view = self.get_view(view_name)
         if view is None:
             return
-        if isinstance(view, QueryIndexView):
-            msg = 'Cannot delete a query index view using this method.'
-            raise CloudantException(msg)
 
         self.views.__delitem__(view_name)
 
     def fetch(self):
         """
         Retrieves the remote design document content and populates the locally
-        cached DesignDocument dictionary.  View content is stored either as
-        View or QueryIndexView objects which are extensions of the ``dict``
+        cached DesignDocument dictionary.  View content is stored as
+        a View object which is an extension of the ``dict``
         type.  All other design document data are stored directly as
         ``dict`` types.
         """
```

Comment:
My original thought was to keep this to not allow adding query indexes in DesignDocument.  But after our earlier discussion, this check should be removed from line [line 119](https://github.com/cloudant/python-cloudant/blob/cefd64747f5ea45ef9782f6776f3977edce817ca/src/cloudant/design_document.py#L119) and [line 138](https://github.com/cloudant/python-cloudant/blob/cefd64747f5ea45ef9782f6776f3977edce817ca/src/cloudant/design_document.py#L138) since CloudantDesignDocument handles all query index logic and other functionality.